teemaderegister-react/src/components/ViewAllUsers.jsx

```diff
@@ -0,0 +1,43 @@
+import React, { useEffect, useState, Component } from 'react'
+import { Layout, Table } from 'antd'
+
+const columns = [
+  {
+    title: 'First Name',
+    dataIndex: 'firstName',
+    key: 'firstName',
+  },
+  {
+    title: 'Last Name',
+    dataIndex: 'lastName',
+    key: 'lastName',
+  },
+]
+
+function ViewAllUsers() {
+  const [users, setUsers] = useState([])
+
+  useEffect(() => {
+    const getUserData = async () => {
+      var jwt = localStorage.getItem('jwtToken')
```

Comment:
Instead of using this, there is an API utils (/util/Api.js) helper that does those things (url + jwt) for you.

Otherwise, You can use this approach, or I would recommend using REDUX as Home pages do. You can keep this component functional and trigger request on useEffect.

Please add loading!

Reply:
There's no such file as Api.js? And what exactly do you mean by "add loading"?

![utils](https://user-images.githubusercontent.com/55079581/102544063-eb430780-40bc-11eb-8fc5-11bc4114c036.png)

Reply:
I don't quite understand how to use the helper. And what do you mean by "add loading"?

Reply:
[CurriculumAdd.jsx](https://github.com/rakenduste-programmeerimine-2020s/teemaderegister/blob/main/teemaderegister-react/src/components/CurriculumAdd.jsx)

There is an example of using the API utils in the "fetchUsers" function

"Add loading" => I'm presuming that you'd have a loading state that is initially false and while it starts fetching the users the loading state is true and will be turned to false again once it's all done. Preferably also showing a spinner while the loading is true.